subprojects/base-services/src/main/java/org/gradle/internal/ImmutableActionSet.java

```diff
@@ -26,9 +26,11 @@
  *
  * Actions are executed in order of insertion. Duplicates are ignored. Execution stops on the first failure.
  *
+ * Implements InternalAction as internal delegates should be directly decorated themselves.
```

Comment:
Good catch! I had to change `ImmutableActionSet` and `MutableActionSet` to implement `InternalListener` instead of `InternalAction` as it caused a package cycle, but missed that I'd mentioned the interface in the javadoc.